furusiya/model/key_binder.py

```diff
@@ -0,0 +1,85 @@
+from model.keys.events import update_callback, quit_event, mousemotion_event
+from model.keys.key_callbacks import *
+
+
+keybinds = {
+    'ESCAPE': exit_to_main_menu_callback,
+    'ENTER': enter_callback,
+
+    'UP': up_callback,
+    'DOWN': down_callback,
+    'LEFT': left_callback,
+    'RIGHT': right_callback,
+
+    'g': pickup_callback,
+
+    'i': inventory_use,
+    'd': inventory_drop,
+
+    'f': bow_callback,
+    'm': mount_callback,
+
+    'r': rest_callback,
+    'R': continuous_rest_callback
+}
+
+events = {
+    'QUIT': quit_event,
+
+    'MOUSEMOTION': mousemotion_event,
+    'MOUSEDOWN': None,
+    'MOUSEUP': None,
+
+    'KEYDOWN': None,
+    'KEYUP': None,
+}
+
+
+class KeyBinder:
+    def __init__(self, game):
+        self.game = game
+
+    # Keybinds
+    def register_all_keybinds(self):
+        for key, callback in keybinds.items():
+            self.register_keybind(key, callback)
+
+    def suspend_all_keybinds(self):
+        for key in keybinds.keys():
+            self.suspend_keybind(key)
+
+    def register_keybind(self, key, callback=None):
+        callback = callback or keybinds.get(key) or (lambda ev: None)
+        setattr(self.game.ui.app, f'key_{key}', callback)
+
+    def suspend_keybind(self, key):
+        delattr(self.game.ui.app, f'key_{key}')
+
+    # Events
+    def register_all_events(self):
+        for key, callback in events.items():
+            self.register_event(key, callback)
+
+        self.register_update()
+
+    def suspend_all_events(self):
```

Comment:
Do we need this?

Reply:
Probably not. I just wrote the entire fleshed out thing and then integrated the keybinds where appropriate.

I'll remove unused ones.